.github/workflows/ci.yml

```diff
@@ -58,21 +58,32 @@ jobs:
         run: cargo test --all-features
       - name: Run test suite with all optimizations
         run: cargo test --release
-      # Do the new project test last, because it adds the crate in the current source
-      # folder, and that shouldn't influence other tests.
+
+  test-init:
```

Comment:
Separated into a isolated step to ensure the isolation that was previously ensured by order, and extracted into a script so it's easier to run locally as well.